app/src/main/java/com/epfl/drawyourpath/database/MockDatabase.kt

```diff
@@ -659,20 +719,100 @@ class MockDatabase : Database() {
             )
         }
         if (tournaments.contains(tournamentId)) {
-            val currentTournament = tournaments[tournamentId]!!
-            tournaments[tournamentId] = currentTournament.copy(
-                participants = currentTournament.participants.filter { it != userId },
+            val currentTournament = tournaments[tournamentId]!!.value!!
+            tournaments[tournamentId]!!.postValue(
+                currentTournament.copy(
+                    participants = currentTournament.participants.filter { it != userId },
+                ),
             )
         }
         return CompletableFuture.completedFuture(Unit)
     }
 
-    override fun getTournament(tournamentId: String): CompletableFuture<Tournament> {
+    override fun getTournament(tournamentId: String): LiveData<Tournament> {
+        if (!tournaments.contains(tournamentId)) {
+            throw Error("This tournament doesn't exist $tournamentId")
+        }
+        return tournaments[tournamentId]!!
+    }
+
+    override fun getTournamentPosts(tournamentId: String): LiveData<List<TournamentPost>> {
+        if (!tournaments.contains(tournamentId)) {
+            throw Error("This tournament doesn't exist $tournamentId")
+        }
+        val postsLiveData = MutableLiveData<List<TournamentPost>>()
+        Handler(Looper.getMainLooper()).post {
+            tournaments[tournamentId]!!.observeForever {
+                postsLiveData.postValue(it.posts)
+            }
+        }
+        return postsLiveData
+    }
+
+    override fun getTournamentParticipantsId(tournamentId: String): CompletableFuture<List<String>> {
+        if (!tournaments.contains(tournamentId)) {
+            return tournamentDoesntExist(tournamentId)
+        }
+        return CompletableFuture.completedFuture(tournaments[tournamentId]!!.value!!.participants)
+    }
+
+    override fun getTournamentInfo(tournamentId: String): LiveData<Tournament> {
+        if (!tournaments.contains(tournamentId)) {
+            throw Error("This tournament doesn't exist $tournamentId")
+        }
+        val infoLiveData = MutableLiveData<Tournament>()
+        Handler(Looper.getMainLooper()).post {
+            tournaments[tournamentId]!!.observeForever {
+                infoLiveData.postValue(it.copy(participants = emptyList(), posts = emptyList()))
+            }
+        }
+        return infoLiveData
+    }
+
+    override fun getPostUID(): String {
+        return mockUID++.toString()
```

Comment:
Same here it 's normal that we use the user id ?

Reply:
Same here: it's not the userID it's a unique id, I agree that it's a bit confusing. I wil rename UID to UniqueId.